spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md

```diff
@@ -193,19 +193,19 @@ public class MyCredentials implements AppConfigCredentialProvider, KeyVaultCrede
 
 ### Modifying Connection Client
 
-In some situations the client connection needs to be modified, such as adding proxy configurations. Using the AppConfigurationClientProvider and/or KeyVaultClientProvider. By implementing either of these classes and providing and generating a @Bean of them will enable client modification following [App Configuration SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/appconfiguration/azure-data-appconfiguration#key-concepts) and [Key Vault SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-secrets#key-concepts).
+In some situations the client connection needs to be modified, such as adding proxy configurations. Using the ConfigurationClientBuilderSetup and/or SecretClientBuilderSetup. By implementing either of these classes and providing and generating a @Bean of them will enable client modification following [App Configuration SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/appconfiguration/azure-data-appconfiguration#key-concepts) and [Key Vault SDK](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-secrets#key-concepts).
 
 ```java
-public class MyClients implements AppConfigurationClientProvider, KeyVaultClientProvider {
+public class MyClients implements ConfigurationClientBuilderSetup, SecretClientBuilderSetup {
 
     @Override
-    public SecretClientBuilder modifyKeyVaultClient(SecretClientBuilder keyVaultClientBuilder) {
-        return keyVaultClientBuilder.httpClient(buildHttpClient());
+    public void setup(SecretClientBuilder builder, String uri) {
+        builder.httpClient(buildHttpClient());
     }
 
     @Override
-    public ConfigurationClientBuilder modifyConfigurationClient(ConfigurationClientBuilder configurationClientBuilder) {
-        return configurationClientBuilder.httpClient(buildHttpClient());
+    public void setup(ConfigurationClientBuilder builder, String endpoint) {
+        builder.httpClient(buildHttpClient());
     }
 
     private HttpClient buildHttpClient() {
```

Comment:
Can the SDK pick up settings `https.proxyHosts` and `https.proxyPort` automatically?

Reply:
It does, but only if we created our own HttpClient like we do in the sample. The issue is that we would either have to choose Netty or Okhttp, which could cause issues with the forced dependency.

Reply:
In the default implementation, isn't the SDK already made the choice of Netty vs. Okhttp?